File: JavaScript/Programmers/Level 1/로또의 최고 순위와 최저 순위.js
// lottos 배열에서 0의 개수와 맞은 번호의 개수를 각각 셈
// 맞은 번호와 0의 개수를 더하면 최고 순위, 맞은 번호가 최저 순위가 됨

function solution(lottos, win_nums) {
    const win = [6, 6, 5, 4, 3, 2, 1]; // 6 ~ 1등까지를 나타내는 배열
    
    const count = lottos.filter(lotto => !lotto).length; // filter 메소드를 이용하여 0인 요소를 필터링 후 길이를 구함, !lotto가 0인 경우를 나타냄
    // filter 메소드를 이용하여 맞은 번호의 요소만 필터링 후 길이를 구함, includes 메소드를 이용하여 번호가 포함되어 있는지 확인
    const answer = lottos.filter(lotto => win_nums.includes(lotto)).length;
    return [win[answer + count], win[answer]]; // answer + count가 최고 순위가 되므로 win 배열에서 해당하는 요소 뽑음, answer가 최저 순위가 되므로 win 배열에서 해당하는 요소 뽑음
}

File: Python/Programmers/Level 1/로또의 최고 순위와 최저 순위.py
# def solution(lottos, win_nums):
#     # 순위 dict
#     rank = {6: 1, 5: 2, 4: 3, 3: 4, 2: 5, 1: 6, 0: 6}
    
#     answer = 0
#     for lotto in lottos:
#         if lotto in win_nums:
#             answer += 1
#     # 최고 순위는 현재 순위에서 0의 개수를 더한 것
#     # 최저 순위는 현재 순위
#     return [rank[answer + lottos.count(0)], rank[answer]]

def solution(lottos, win_nums):
    rank = [6, 6, 5, 4, 3, 2, 1] # 순위 리스트
    answer = len(set(win_nums) & set(lottos)) # 교집합을 이용하여 현재 순위를 구함
    return [rank[answer + lottos.count(0)], rank[answer]]
